File: WADV.CGModule/PluginInterface/Initialiser.cs
using System;
using WADV.Core.API;
using WADV.CGModule.API;
using WADV.Core.PluginInterface;

namespace WADV.CGModule.PluginInterface
{
    internal sealed class PluginInitialise : IPluginInitialise
    {

        public bool Initialising()
        {
            ScriptAPI.RegisterInTableSync("api_cg", "init", new Action<int>(ConfigAPI.Init), true);
            ScriptAPI.RegisterInTableSync("api_cg", "show", new Func<string, string, int, string, bool>(ImageAPI.Show));
            ScriptAPI.RegisterInTableSync("api_cg", "clean", new Func<string, bool>(ImageAPI.Clean));
            MessageAPI.SendSync("[CG]INIT_LOAD_FINISH");
            return true;
        }

    }
}

File: WADV.ChoiceModule/PluginInterface/Initialiser.vb
Imports System.Windows.Controls
Imports WADV.Core.PluginInterface
Imports WADV.ChoiceModule.API

Namespace PluginInterface

    Friend NotInheritable Class PluginInitialise : Implements IPluginInitialise

        Public Function Initialising() As Boolean Implements IPluginInitialise.Initialising
            ScriptAPI.RegisterInTableSync("choice", "init", New Action(Of String, String, Double, Integer)(AddressOf ConfigAPI.Init), True)
            ScriptAPI.RegisterInTableSync("choice", "setContent", New Action(Of Panel)(AddressOf ConfigAPI.SetContent))
            ScriptAPI.RegisterInTableSync("choice", "setStyle", New Action(Of String)(AddressOf ConfigAPI.SetStyle))
            ScriptAPI.RegisterInTableSync("choice", "setMargin", New Action(Of Double)(AddressOf ConfigAPI.SetMargin))
            ScriptAPI.RegisterInTableSync("choice", "setZIndex", New Action(Of Integer)(AddressOf ConfigAPI.SetZIndex))
            ScriptAPI.RegisterInTableSync("choice", "show", New Func(Of Neo.IronLua.LuaTable, String, String, Integer, String, String)(AddressOf ChoiceAPI.ShowByLua))
            MessageAPI.SendSync("[CHOICE]INIT_LOAD_FINISH")
            Return True
        End Function

    End Class

End Namespace
